addons/mail/static/src/chatter/web/mail_composer_form.js

```diff
@@ -27,6 +31,24 @@ export class MailComposerFormRenderer extends formView.Renderer {
             this.root,
             this.props.record.resId
         ]);
+
+        // Add file dropzone on full mail composer:
+        this.attachmentUploadService = useService("mail.attachment_upload");
+        this.operations = useX2ManyCrud(() => {
+            return this.props.record.data["attachment_ids"];
+        }, true);
+
+        useDropzone(this.root, async event => {
+            const resIds = JSON.parse(this.props.record.data.res_ids);
+            const thread = await this.mailStore.Thread.getOrFetch({
+                model: this.props.record.data.model,
+                id: resIds[0],
+            });
+            for (const file of event.dataTransfer.files) {
+                const attachment = await this.attachmentUploadService.upload(thread, thread.composer, file);
```

Comment:
`getOrFetch` is currently only useful for channels (compared to just using `get`) but it doesn't hurt either to use it.

Also it's not a guaranteed find, so we usually ensure the thread was found before going further.

If you are guaranteed the thread exists (so model and id are not user input, and the thread could not have been deleted in the meantime), then you could use `insert` instead to ensure you have it and to avoid having to check existence.